python/pyarrow/__init__.py

```diff
@@ -262,8 +262,11 @@ def append_library_dir(library_dir):
         if _has_pkg_config(pkgname):
             library_dir = _read_pkg_config_variable(pkgname,
                                                     ["--libs-only-L"])
-            assert library_dir.startswith("-L")
-            append_library_dir(library_dir[2:])
+            # pkg-config output could be empty if Arrow is installed
+            # as a system package.
+            if library_dir:
+                assert library_dir.startswith("-L")
```

Comment:
This is not test code, so we should rather raise than `assert`?

Otherwise LGTM